### PR TITLE
fix: Update numpy deprecation of product and test dependency issues

### PR DESCRIPTION
**Pull request by cssherman**
Temporary testing branch: https://github.com/GEOS-DEV/GEOS/pull/3328

### CODE REVIEW

**Comment by cssherman** [comment]
@rrsettgast - it looks like this fixed the deprecation issue on the CI machine

File: geos-ats/src/geos/ats/helpers/curve_check.py [hunk 1]
```diff
@@ -33,16 +33,18 @@ def interpolate_values_time( ta, xa, tb ):
     """
     N = list( np.shape( xa ) )
     M = len( tb )
+    ta = np.ascontiguousarray( np.squeeze( ta ) )
+    tb = np.ascontiguousarray( np.squeeze( tb ) )
```
Comment (cssherman) [inline geos-ats/src/geos/ats/helpers/curve_check.py:37]: I think that the issue we ran into was due to these time vectors changing shape from (N,) to (N, 1) was the issue that triggered the interpolation issue.

File: geos-ats/src/geos/ats/helpers/curve_check.py [hunk 2]
```diff
@@ -33,16 +33,18 @@ def interpolate_values_time( ta, xa, tb ):
...
         return interp1d( ta, xa )( tb )
     else:
         # Reshape the input array so that we can work on the non-time axes
-        S = np.product( N[ 1: ] )
+        S = np.prod( N[ 1: ] )
         xc = np.reshape( xa, ( N[ 0 ], S ) )
         xd = np.zeros( ( len( tb ), S ) )
         for ii in range( S ):
-            xd[ :, ii ] = interp1d( ta, xc[ :, ii ] )( tb )
+            xd[ :, ii ] = interp1d( ta, xc[ :, ii ], bounds_error=False, fill_value='extrapolate' )( tb )
```
Comment (cssherman) [inline geos-ats/src/geos/ats/helpers/curve_check.py:47]: It's often not great, but I'm allowing the interpolator to extrapolate the comparison here (there will always be an error in this case due to changes in the time vector though... this is purely to check how much the curves changed if they were on the same time basis)

File: geos-ats/src/geos/ats/helpers/curve_check.py [hunk 3]
```diff
@@ -33,16 +33,18 @@ def interpolate_values_time( ta, xa, tb ):
...
     M = len( tb )
+    ta = np.ascontiguousarray( np.squeeze( ta ) )
+    tb = np.ascontiguousarray( np.squeeze( tb ) )
 
     if ( len( N ) == 1 ):
         return interp1d( ta, xa )( tb )
     else:
         # Reshape the input array so that we can work on the non-time axes
-        S = np.product( N[ 1: ] )
+        S = np.prod( N[ 1: ] )
```
Comment (cssherman) [inline geos-ats/src/geos/ats/helpers/curve_check.py:43]: This fixes the numpy.product deprecation

**Review by CusiniM — APPROVED** [review]

**Pull request merged**